libraries/AP_PiccoloCAN/AP_PiccoloCAN.cpp

```diff
@@ -690,22 +691,28 @@ bool AP_PiccoloCAN::handle_esc_message(AP_HAL::CANFrame &frame)
         // There are no common error bits between the Gen-1 and Gen-2 ICD
     } else if (decodeESC_StatusBPacket(&frame, &esc.voltage, &esc.current, &esc.dutyCycle, &esc.escTemperature, &esc.motorTemperature)) {
         
-        AP_ESC_Telem_Backend::TelemetryData t {
-            .temperature_cdeg = int16_t(esc.escTemperature * 100),
-            .voltage = float(esc.voltage) * 0.01f,
-            .current = float(esc.current) * 0.01f,
-        };
+        AP_ESC_Telem_Backend::TelemetryData telem = {};
 
-        update_telem_data(addr, t,
+        telem.voltage = float(esc.voltage) * 0.01f;
+        telem.current = float(esc.current) * 0.01f;
+        telem.motor_temp_cdeg = int16_t(esc.motorTemperature * 100);
+
+        update_telem_data(addr, telem,
             AP_ESC_Telem_Backend::TelemetryType::CURRENT
                 | AP_ESC_Telem_Backend::TelemetryType::VOLTAGE
                 | AP_ESC_Telem_Backend::TelemetryType::TEMPERATURE);
 
         esc.newTelemetry = true;
     } else if (decodeESC_StatusCPacket(&frame, &esc.fetTemperature, &esc.pwmFrequency, &esc.timingAdvance)) {
-        AP_ESC_Telem_Backend::TelemetryData t { };
-        t.motor_temp_cdeg = int16_t(esc.fetTemperature * 100);
-        update_telem_data(addr, t, AP_ESC_Telem_Backend::TelemetryType::MOTOR_TEMPERATURE);
+
+        // Use the higher reported value of 'escTemperature' and 'fetTemperature'
+        const int16_t escTemp = MAX(esc.fetTemperature, esc.escTemperature);
+
+        AP_ESC_Telem_Backend::TelemetryData telem = {};
```

Comment:
```suggestion
        AP_ESC_Telem_Backend::TelemetryData telem {};
```

.... usually the compiler will elide this sort of thing - but if it doesn't what you end up doing is contructing an `AP_ESC_Telem_Backend::TelemetryData` on the stack and then copying it back over the `telem` `AP_ESC_Telem_Backend::TelemetryData`.  Omitting the ` =` avoids this.